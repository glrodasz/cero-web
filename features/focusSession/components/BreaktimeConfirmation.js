import PropTypes from 'prop-types'

import {
  Modal,
  CenteredContent,
  Picture,
  Heading,
  Spacer,
  Paragraph,
  Button,
} from '@glrodasz/components'

import { time } from '../../common/constants'

const handleClickChoose = ({ onClickChoose }) => (time) => () => {
  onClickChoose(time)
}

const BreaktimeConfirmation = ({ onClickClose, onClickChoose }) => {
  const handleChooseBreaktime = handleClickChoose({ onClickChoose })

  return (
    <Modal isCentered onClose={onClickClose}>
      <CenteredContent>
        <Picture src="/images/couch-pause.svg" width={200}></Picture>
        <Spacer.Vertical size="md" />
        <Heading size="xl" color="tertiary">
          Tomate un tiempo para refrescarte
        </Heading>
        <Spacer.Vertical size="sm" />
        <Paragraph>
          Siempre hay que celebrar los peque√±os triunfos, por eso te invitamos a
          tomar un descanso para despejar tu mente.
        </Paragraph>
        <Spacer.Vertical size="md" />
        <div style={{ display: 'flex', gap: '0 20px' }}>
          <Button
            onClick={handleChooseBreaktime(time.FIVE_MINUTES_IN_MS)}
            isMuted
          >
            5 min
          </Button>
          <Button
            onClick={handleChooseBreaktime(time.TEN_MINUTES_IN_MS)}
            isMuted
          >
            10 min
          </Button>
          <Button
            onClick={handleChooseBreaktime(time.FIFTY_MINUTES_IN_MS)}
            isMuted
          >
            15 min
          </Button>
        </div>
      </CenteredContent>
    </Modal>
  )
}

BreaktimeConfirmation.propTypes = {
  onClickClose: PropTypes.func.isRequired,
  onClickChoose: PropTypes.func.isRequired,
}

export default BreaktimeConfirmation
