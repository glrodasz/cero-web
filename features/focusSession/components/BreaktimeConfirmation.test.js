import BreaktimeConfirmation from './BreaktimeConfirmation'
import { render } from '@testing-library/react'

jest.mock('@glrodasz/components', () => {
  const { dummyRender } = require('../../../utils/tests/dummyRender')
  return {
    Modal: dummyRender('Modal'),
    CenteredContent: dummyRender('CenteredContent'),
    Picture: dummyRender('Picture'),
    Heading: dummyRender('Heading'),
    Spacer: { Vertical: dummyRender('Spacer.Vertical') },
    Paragraph: dummyRender('Paragraph'),
    Button: dummyRender('Button'),
  }
})

describe('[ features / focusSession / components / BreaktimeConfirmation ]', () => {
  describe('when `BreaktimeConfirmation` is mounted', () => {
    it('should render', () => {
      const props = {
        onClickClose: () => {},
        onClickChoose: () => {},
      }
      const { asFragment } = render(<BreaktimeConfirmation {...props} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
