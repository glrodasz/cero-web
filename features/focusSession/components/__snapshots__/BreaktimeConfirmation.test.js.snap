// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ features / focusSession / components / BreaktimeConfirmation ] when \`BreaktimeConfirmation\` is mounted should render 1`] = `
<DocumentFragment>
  [Modal {"isCentered":true}][CenteredContent][Picture {"src":"/images/couch-pause.svg","width":200} /] [Spacer {"size":"md"} /] [Heading {"size":"xl","color":"tertiary"}]Tomate un tiempo para refrescarte[/Heading] [Spacer {"size":" sm"} /] [Paragraph]Siempre hay que celebrar los peque√±os triunfos, por eso te invitamos a tomar un descanso para despejar tu mente.[/Paragraph] [Spacer {"size":"md"} /] [div style="display:flex;gap:0 20px"][Button {"isMuted":true}]5 min[/Button] [Button {"isMuted":true}]10 min[/Button] [Button {"isMuted":true}]15 min[/Button] [/div][/CenteredContent] [/Modal] 
</DocumentFragment>
`;
